ts/src/pro/bingx.ts

```diff
@@ -224,6 +237,206 @@ export default class bingx extends bingxRest {
+        const request: Dict = {
+            'id': uuid,
+            'dataType': 'all@ticker',
+        };
+        if (marketType === 'swap') {
+            request['reqType'] = 'sub';
+        }
+        const result = await this.watchMultiple (url, messageHashes, this.deepExtend (request, params), subscriptionHashes);
+        // for efficiency, we have two type of returned structure here - if symbols array was provided, then individual
+        // ticker dict comes in, otherwise all-tickers dict comes in
+        if (this.newUpdates) {
+            const newDict: Dict = {};
+            newDict[result['symbol']] = result;
+            return newDict;
+        }
+        return this.tickers;
+    }
+
+    async watchOrderBookForSymbols (symbols: string[], limit: Int = undefined, params = {}): Promise<OrderBook> {
+        /**
+         * @method
+         * @name bingx#watchOrderBookForSymbols
+         * @description watches information on open orders with bid (buy) and ask (sell) prices, volumes and other data
+         * @see https://bingx-api.github.io/docs/#/en-us/swapV2/socket/market.html#Subscribe%20Market%20Depth%20Data%20of%20all%20trading%20pairs
+         * @param {string[]} symbols unified array of symbols
+         * @param {int} [limit] the maximum amount of order book entries to return
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {object} A dictionary of [order book structures]{@link https://docs.ccxt.com/#/?id=order-book-structure} indexed by market symbols
+         */
+        symbols = this.marketSymbols (symbols, undefined, true, true, false);
+        let firstMarket = undefined;
+        let marketType = undefined;
+        const symbolsDefined = (symbols !== undefined);
+        if (symbolsDefined) {
+            firstMarket = this.market (symbols[0]);
+        }
+        [ marketType, params ] = this.handleMarketTypeAndParams ('watchOrderBookForSymbols', firstMarket, params);
+        if (marketType === 'spot') {
+            throw new NotSupported (this.id + ' watchOrderBookForSymbols is not supported for spot markets yet');
+        }
+        limit = this.getOrderBookLimitByMarketType (marketType, limit);
+        let interval = undefined;
+        [ interval, params ] = this.handleOptionAndParams (params, 'watchOrderBookForSymbols', 'interval', 500);
+        this.checkRequiredArgument ('watchOrderBookForSymbols', interval, 'interval', [ 100, 200, 500, 1000 ]);
+        const channelName = 'depth' + limit.toString () + '@' + interval.toString () + 'ms';
+        const subscriptionHash = 'all@' + channelName;
+        const messageHashes = [];
+        if (symbolsDefined) {
+            for (let i = 0; i < symbols.length; i++) {
+                const symbol = symbols[i];
+                const market = this.market (symbol);
+                messageHashes.push (this.getMessageHash ('orderbook', channelName, market['symbol']));
```

Comment:
again, why make the `channel name` part of the messages? I thought we discussed this

Reply:
i think i've kept that bcz some methods (ohlcv,etc) needed channelname (1min, 5min, etc..). give me few mins, i'll correct them